### PR TITLE
Feature: fathom chart events

### LINKED ISSUE #1674: Fathom events - 18.1
- Trigger an event everytime user clicks Pro
- Trigger an event everytime user clicks Off
- Differentiate volume from market orders and limit orders for CoW
- Time spent on Pro mode? How many users spend more than 15 seconds, so everytime we refresh trade history, send an event with a counter. 
- How many times user updates Limit order price? And at what % above or below user places the limit order?

### PR DESCRIPTION
**Pull request by berteotti**
# Summary

Fixes #1674

Adds fathom charts events.

  # To Test

1. Do some trades while in pro mode in different chains
- [ ] Verify that event for that trade appears like: `<chain>/ecoRouter/<exchange>/volumeUSD/pro`

### CODE REVIEW

**Comment by MilanVojnovic95** [comment]
Tested and nothing changes on https://app.usefathom.com/share/uksjthlz/swapr+%28staging%29

I did Swap in pro mode and these are the transactions I made on different networks
image

And this is a situation that was present even before these transactions
image

**Comment by MilanVojnovic95** [comment]
@berteotti 
Tested and one value that increases is the only one that is marked
image

**Comment by berteotti** [comment]
> @berteotti Tested and one value that increases is the only one that is marked image

Just added some example test so you can see if it applies. I think it's not working when trading with native tokens, but works fine otherwise, can you test it again please?

**Comment by MilanVojnovic95** [comment]
@berteotti 
Tested and looks ok to me, the value I marked increases when swapping a native token like the others  👍

File: src/analytics/provider/analytics.provider.tsx [hunk 1]
```diff
@@ -128,24 +133,25 @@ export function AnalyticsProvider({ children }: AnalyticsProviderProps) {
...
+        const id = computeItemId(trade, chartOption)
         setTradeQueue(state => ({
           ...state,
           [id]: {
             id,
             item: trade,
+            chartOption,
             status: ItemStatus.PENDING,
             retries: 0,
           },
```
Comment (bejzik8) [inline src/analytics/provider/analytics.provider.tsx:157]: I get that this is an old code but I think it would be much cleaner with `async` / `await`.

**Review by bejzik8 — APPROVED** [review]
Code looks good. 👍🏼

**Pull request merged**